### ISSUE TITLE
类似于语义角色标注的sequence 2 sequence任务，但label形式不一样，该如何处理

### ISSUE THREAD

**Issue by gjworm** [PaddlePaddle/Paddle#3909]
任务类型和示例6的语义角色标注形式上基本一样，语义角色标注里，sequence形式的label是[int, int, int, ...]，每个int的范围是0 - label_dict_len， 最后
```python
crf_cost = paddle.layer.crf(
    size=label_dict_len,
    input=feature_out,
    label=target,
    param_attr=paddle.attr.Param(
        name='crfw',
        initial_std=default_std,
        learning_rate=mix_hidden_lr))
```
crf_cost的size是每个时刻label的取值范围。
但我的任务里，sequence形式的label是`[[0 0 0 1  1 0], [1 1 0 0 1 0], [1 0 0 1 0 1]....]`， 也就是每个时刻不是输入一个数字，而是多个数字，这样的情况cost层该如何设置呢

**Comment by lcy-seso** [comment]
多个数字代码什么含义？想如何计算cost？

**Comment by gjworm** [comment]
这个任务大概可以理解为序列图像解析，也就是有一串时序上连续的图片序列，每张图片每个像素有一个label（0或1），前后帧间各像素label是有关联的，训练时送进去图片序列和label图的序列，预测时通过送进去的图片序列得到预测的label图序列。

**Comment by gjworm** [comment]
@lcy-seso 我觉得应该挺适合用crf_cost，但如果不支持这种形式，用交叉熵也可以，但我看cross_entropy_cost又好像不支持序列输入，就不知该怎么配置了

**Comment by lcy-seso** [comment]
- crf 基于整个序列计算cost，用于序列标注，可以用。你这个 0 - 1 没有解释的含义，你可以试着将问题转化成序列标注问题用crf。
- 机器翻译里面的交叉熵接受的就是序列输入。交叉熵不关心输入是否是序列。

**Comment by gjworm** [comment]
如果这个问题转化成序列标注，那label_dict_len就太长了。我试试交叉熵吧，非常感谢！

**Comment by lcy-seso** [comment]
现在只有 0- 1 你怎么考虑转化的？

**Comment by gjworm** [comment]
每个时刻输入的是一张图，假如是10*10的，每张图每个像素的label组合起来的可能性就有2^100种可能，那label_dict_len就得有2^100这么长吧

**Comment by lcy-seso** [comment]
当然不能这样转化，我不知道你这个 0-1的具体含义，序列标注本身有分隔某个有意义片段这样的作用。

**Comment by gjworm** [comment]
哦，这样的话是不一样，不太了解NLP，看那个例子像给每个词赋一个label。我的任务就是给序列图片中每个像素赋一个label，比如表示该像素是前景还是背景。

**Comment by lcy-seso** [comment]
那直接用 cross_entroy 跑一个base line，如果需要基于全序列计算 cost ，需要一些设计暂时没有。

**Comment by gjworm** [comment]
ok，多谢

**Comment by lcy-seso** [comment]
那我先关闭这个issue，如果有新的问题，可以随时 reopen。或者建立新的 issue。非常感谢。

**Comment by gjworm** [comment]
@lcy-seso 我试了个简单的配置，但train的时候总是出错，我看了些相关的issue和例子，觉得有可能是label形式有问题，但我也试过integer_value_sequence也不行（这个形式应该不符合我的需求），下面是简单的配置文件，能看看是什么问题吗
    label_len = 300
    def d_type(size):
        return paddle.data_type.dense_vector_sequence(size)

    target = paddle.layer.data(name='target', type=d_type(label_len))
    x = paddle.layer.data(name='x', type=d_type(label_len))

    hidden1 = paddle.layer.fc(input=x, size=200, act=linear)
    lstm1 = paddle.layer.lstmemory(input=hidden1, act=relu)
    feature_out = paddle.layer.fc(input=[hidden1, lstm1], size=label_len, act=sigmoid)

    cross_entropy_cost = paddle.layer.multi_binary_label_cross_entropy_cost(input=feature_out, label=target)

    parameters = paddle.parameters.create(cross_entropy_cost)
    optimizer = paddle.optimizer.Momentum(momentum=0.5, learning_rate=0.1)

    trainer = paddle.trainer.SGD(
    cost=cross_entropy_cost,
    parameters=parameters,
    update_equation=optimizer)

读入第一条数据后，就出
F0906 12:06:44.781072     1 Matrix.cpp:3271] Check failed: dynamic_cast<CpuIVector*>(&label)
这样的错误

**Comment by gjworm** [comment]
cost是cross_entropy_cost，不是multi_binary_label_cross_entropy_cost

**Comment by lcy-seso** [comment]
cross_entropy 的label 对应的type是 integer_value_sequence.

**Comment by gjworm** [comment]
我理解的integer_value_sequence应该是整数序列吧，在这里面label应该是vector的vector

**Comment by gjworm** [comment]
我用mse就可以用dense_vector_sequence了，cross_entropy是不支持这样吗？还有batch size稍微大点，就会训几轮就挂了，是不是因为lstm占内存太大？
